### ISSUE TITLE
Allow setting AutoRemove (`--rm`) by default in `config.json`

### ISSUE THREAD

**Issue by toothbrush** [docker/cli#4763]
### Description

As a user, i'd like to be able to set something like `"autoRemove": true` in my `~/.docker/config.json` file, so that i can run a command as follows:

```
docker run -it debian
```

...and have it be equivalent to:

```
docker run --rm -it debian
```

Because i always forget the `--rm` flag, and end up with hundreds of dangling containers in my Docker dashboard.

I'm not proposing changing the default, just allowing me a persistent way to set the flag.  I've had a look at the source and i don't believe there's any environment variable that i could set that would have this effect.

I'll open a small PR that enables the feature i suggest, but i'm sure it won't be ready as-is, because i'll have forgotten cases or whatever, but i'd like to have a starting point for the discussion.

**Comment by vvoland** [comment]
Possibly it could also be useful if this would be generalized this to support all options, for example:

```json
# config.json
{
  ...
  "defaultContainerOpts": {
     "rm": true,
     "interactive": true,
     "tty": true
  }
}
```

Which would mean that `docker run ...` would now do the same as `docker run --rm -it ...`.
Although, that could also become a big footgun, so I'm not sure if we should actually do that.

**Comment by toothbrush** [comment]
> Possibly it could also be useful if this would be generalized this to support all options

Good point – this is exactly why i threw up a strawman PR because i was sure the maintainers would have better suggestions :+1:  

> that could also become a big footgun

For sure.  The power user in me likes having the flexibility and to be honest the machinery around CLI config.json looks a bit anemic / forgotten, so i'd personally argue to expose the options as you describe.  Plus, i'd imagine if someone is putting these flags in their config.json, that presumably self-selects for folks who know they're tinkering.. but i definitely would understand if my point of view is too optimistic!

**Comment by toothbrush** [comment]
How about something like this:

```json
# config.json
{
  ...
  "containerOpts": {
     "rm": true,
     "interactive": true,
     "tty": true
  }
}
```

Naming is hard 😅 Maybe `containerOptions` or just `container` is better than `containerOpts`?
